src/runtime/allocator.ts

```diff
@@ -0,0 +1,291 @@
+
+    const partitionByFactory = new Map<ArcHostFactory, Particle[]>();
+    // Partition the `plan` into particles by ArcHostFactory.
+    for (const particle of plan.particles) {
+      const hostFactory = [...this.arcHostFactories.values()].find(
+          factory => factory.isHostForParticle(particle));
+      assert(hostFactory);
+      if (!partitionByFactory.has(hostFactory)) {
+        partitionByFactory.set(hostFactory, []);
+      }
+      partitionByFactory.get(hostFactory).push(particle);
+    }
+
+    // Start all partitions.
+    return Promise.all([...partitionByFactory.keys()].map(async factory => {
+      const host = factory.createHost();
+      this.hostById[host.hostId] = host;
+
+      const partial = newRecipe();
+      plan.mergeInto(partial);
+
+      const partitionParticles = partitionByFactory.get(factory);
+      plan.particles.forEach((particle, index) => {
+        if (!partitionParticles.find(p => p.name === particle.name)) {
+          plan.particles.splice(index, 1);
+        }
+      });
+      await this.assignStorageKeys(arcId, partial);
+
+      const arcOptions = {arcId, idGenerator: this.arcStateById.get(arcId).idGenerator};
+      const partition = {arcHostId: host.hostId, arcOptions, plan: partial, reinstantiate};
+      this.arcStateById.get(arcId).partitions.push(partition);
+
+      return host.start(partition);
+    }));
+  }
+
+  async assignStorageKeys(arcId: ArcId, plan: Recipe, idGenerator?: IdGenerator): Promise<Recipe> {
+    // TODO(b/182410550): All internal caller(s) should pass non normalized recipe.
+    // Remove this check, once the method is private, and don't return recipe.
+    if (plan.isNormalized()) {
+      plan = plan.clone();
+    }
+    // Assign storage keys for all `create` & `copy` stores.
+    for (const handle of plan.handles) {
+      if (handle.immediateValue) continue;
+      if (['copy', 'create'].includes(handle.fate)) {
+      let type = handle.type;
+        if (handle.fate === 'create') {
+          assert(type.maybeResolve(), `Can't assign resolved type to ${type}`);
+        }
+
+        type = type.resolvedType();
+        assert(type.isResolved(), `Can't create handle for unresolved type ${type}`);
+        handle.id = handle.fate === 'create' && !!handle.id
+          ? handle.id
+          : (idGenerator || this.arcStateById.get(arcId).idGenerator).newChildId(arcId, '').toString();
+        handle.fate = 'use';
+        handle.storageKey = await this.runtime.getCapabilitiesResolver(arcId)
+          .createStorageKey(handle.capabilities || Capabilities.create(), type, handle.id);
+      }
+    }
+    return this.resolveRecipe(arcId, plan);
+  }
+
+  // Returns the resolved recipe (the original recipe might have changed),
+  // or throws an exception, if the recipe cannot be resolved.
+  // Note this method is overriden by the subclass, and calls RecipeResolver, if needed.
+  // The APIs might change, when Planning is incorporated in Allocator.
```

Comment:
I don't think we want Planning in the Allocator; I think it makes sense for it to be a separate component (as in they don't need to share data so they don't need to be co-located)

Reply:
these methods were removed.